livekit-agents/livekit/agents/pipeline/agent_output.py

```diff
@@ -170,8 +171,11 @@ async def _synthesize_task(self, handle: SynthesisHandle) -> None:
 
         if isinstance(tts_source, Awaitable):
             tts_source = await tts_source
-            co = _str_synthesis_task(tts_source, transcript_source, handle)
-        elif isinstance(tts_source, str):
+        if isinstance(transcript_source, Awaitable):
+            transcript_source = await transcript_source
+
+        if isinstance(tts_source, str):
+            assert isinstance(transcript_source, str)
```

Comment:
I don't think this would ever happen (tts_source is str, transcript is stream) but could implement it if we wanted.